src/passes/Outlining.cpp

```diff
@@ -13,23 +13,238 @@
 #define DBG(statement)
 #endif
 
+// Check a Result or MaybeResult for error and call Fatal() if the error exists.
+#define ASSERT_OK(val)                                                         \
+  if (auto _val = (val); auto err = _val.getErr()) {                           \
+    Fatal() << err->msg;                                                       \
+  }
+
 namespace wasm {
 
+// Instances of this walker are intended to walk a function at a time, at the
+// behest of the owner of the instance.
+struct ReconstructStringifyWalker
+  : public StringifyWalker<ReconstructStringifyWalker> {
+
+  ReconstructStringifyWalker(Module* wasm)
+    : existingBuilder(*wasm), outlinedBuilder(*wasm) {
+    this->setModule(wasm);
+    DBG(std::cout << "\n\nexistingBuilder: " << &existingBuilder
+                  << " outlinedBuilder: " << &outlinedBuilder);
+  }
+
+  // As we reconstruct the IR during outlining, we need to know what
+  // state we're in to determine which IRBuilder
+  // to send the instruction.
```

Comment:
```suggestion
  // to send the instruction to.
```
Also this could be re-wrapped to 80 cols.

Reply:
done